src/middleware.js

```diff
@@ -23,22 +23,31 @@ const verifyToken = (req, res, next) => {
 		json: true,
 	};
 
-    request.get(authOptions, (err, res, body) => {
-        if (body.error && res.statusCode === 401) {
-            getAccessToken(refreshToken)
-            .then( token => {
-                req.accessToken = token
-                next()
-            })
-            .catch( err => {
-                throw err.message
-            })
-        } else {
-            req.accessToken = accessToken
-            next()
-        }
-    })
-}
+	request.get(authOptions, (err, res, body) => {
+		if (body.error) {
+			getAccessToken(refreshToken)
+				.then((token) => {
+					let authOptions = {
+						url: "https://api.spotify.com/v1/me",
```

Comment:
This could be shortened a little to modifying `authOptions` instead of reassigning it.